File: archive/g/go/palindromic-number.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func palindromicNumber(x int) {
	if x >= 0 {
		reversedNumber := 0
		noOfDigits := 0
		temp := x

		for temp > 0 {
			noOfDigits++
			reversedNumber = (reversedNumber * 10) + (temp % 10)
			temp /= 10
		}

		if x == reversedNumber {
			fmt.Println("true")
		} else {
			fmt.Println("false")
		}
	} else {
		fmt.Println("Usage: please input a non-negative integer")
	}
}

func main() {
	if len(os.Args) < 2 {
		fmt.Println("Usage: please input a non-negative integer")
		os.Exit(1)
	}

	x, err := strconv.Atoi(os.Args[1])
	if err != nil {
		fmt.Println("Usage: please input a non-negative integer")
		os.Exit(1)
	}

	palindromicNumber(x)
}

File: archive/e/elixir/palindromic-number.ex
defmodule PalindromicNumber do
  @doc """
  Checks whether the first argument from the args list is a palindromic number

  Returns either `{:ok, result}` or `:error` when input is empty or not a non-negative integer
  """
  @spec main(argv :: list(String.t())) :: {:ok, boolean()} | :error
  def main([number]) do
    with {number, ""} when number >= 0 <- Integer.parse(number) do
      {:ok, palindromic?(number)}
    else
      _ -> :error
    end
  end
  def main(_), do: :error

  @spec palindromic?(number :: integer()) :: boolean()
  defp palindromic?(number), do: reverse_number(number) == number

  @spec reverse_number(number :: integer(), acc :: integer()) :: integer()
  defp reverse_number(number, acc \\ 0)
  defp reverse_number(0, acc), do: acc
  defp reverse_number(number, acc), do: reverse_number(div(number, 10), acc * 10 + rem(number, 10))
end

case PalindromicNumber.main(System.argv()) do
  {:ok, res} -> IO.puts(res)
  :error     -> IO.puts("Usage: please input a non-negative integer")
end
